// shrinkpack/dist/lib/log.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
exports.log = void 0;
var picocolors_1 = __importDefault(require("picocolors"));
exports.log = {
    deletion: function (msg) {
        console.log(picocolors_1["default"].red('-'), msg);
    },
    download: function (msg) {
        console.log(picocolors_1["default"].yellow('↓'), msg);
    },
    error: function (msg, err) {
        console.error(picocolors_1["default"].red("! ".concat(msg)));
        if (isError(err)) {
            console.error(picocolors_1["default"].dim(err.stack || ''));
        }
    },
    info: function (msg) {
        console.log(picocolors_1["default"].blue("i ".concat(msg)));
    },
    verbose: function (msg) {
        if (process.env.SHRINKPACK_LOG_LEVEL === 'verbose') {
            console.error(picocolors_1["default"].gray("# ".concat(msg)));
        }
    }
};
function isError(value) {
    return Boolean(value && typeof value === 'object' && 'stack' in value);
}
